txpipe/extensions/twopoint_scia.py

```diff
@@ -0,0 +1,493 @@
+                    patch_centers=patch_centers)
+                    #npatch=self.config['npatch'])
+            else:
+                cat = treecorr.Catalog(
+                    g1 = g1,
+                    g2 = g2,
+                    r = r,
+                    ra = data['ra'][mask],
+                    dec = data['dec'][mask],
+                    ra_units='degree', dec_units='degree')
+        else:
+            if self.config['var_method']=='jackknife':
+                patch_centers = self.get_input('patch_centers')
+                cat = treecorr.Catalog(
+                    g1 = g1,
+                    g2 = g2,
+                    ra = data['ra'][mask],
+                    dec = data['dec'][mask],
+                    ra_units='degree', dec_units='degree',
+                    patch_centers=patch_centers)
+                    #npatch=self.config['npatch'])
+            else:
+                cat = treecorr.Catalog(
+                    g1 = g1,
+                    g2 = g2,
+                    ra = data['ra'][mask],
+                    dec = data['dec'][mask],
+                    ra_units='degree', dec_units='degree')
+
+        return cat
+
+    def calculate_shear_pos_select(self,data, meta, i, j):
+        # This is the added calculation that uses the selection function, as defined in our paper. it picks
+        # out all the pairs where the object in the source catalog is in front of the lens object. 
+        # note we are looking at auto-correlations for the source bins!
+        import treecorr 
+
+        cat_i = self.get_shear_catalog(data, meta, i)
+        n_i = cat_i.nobj
+
+        cat_j, rancat_j = self.get_lens_catalog(data, meta, j)
+        n_j = cat_j.nobj
+        n_rand_j = rancat_j.nobj if rancat_j is not None else 0
+
+        print(f"Rank {self.rank} calculating shear-position bin pair ({i},{j}): {n_i} x {n_j} objects, {n_rand_j} randoms")
+
+        ng = treecorr.NGCorrelation(self.config, max_rpar = 0.0)
+        ng.process(cat_j, cat_i)
+
+        if rancat_j:
+            rg = treecorr.NGCorrelation(self.config, max_rpar = 0.0)
+            rg.process(rancat_j, cat_i)
+        else:
+            rg = None
+
+        ng.calculateXi(rg=rg)
+
+        return ng
+
+    def calculate_shear_pos(self, data, meta, i, j):
+        import treecorr
+
+        cat_i = self.get_shear_catalog(data, meta, i)
+        n_i = cat_i.nobj
+
+        cat_j, rancat_j = self.get_lens_catalog(data, meta, j)
+        n_j = cat_j.nobj
+        n_rand_j = rancat_j.nobj if rancat_j is not None else 0
+
+        print(f"Rank {self.rank} calculating shear-position bin pair ({i},{j}): {n_i} x {n_j} objects, {n_rand_j} randoms")
+
+        ng = treecorr.NGCorrelation(self.config)
+        ng.process(cat_j, cat_i)
+
+        if rancat_j:
+            rg = treecorr.NGCorrelation(self.config)
+            rg.process(rancat_j, cat_i)
+        else:
+            rg = None
+
+        ng.calculateXi(rg=rg)
+
+        return ng
```

Comment:
two identical pieces of code except for 

rg = treecorr.NGCorrelation(self.config, max_rpar = 0.0)

there is probably a way to combine these and still get what you want, but no big deal I leave this up to you.  However, I think it is worth explaining by a brief comment why  max_rpar = 0.0 does the trick!

Reply:
The max_rpar does a huge difference, could probably be done with a flag and some if statements, but figured it was easier to leave it as 2 different functions. 

added.